vertx-core/src/main/java/io/vertx/core/http/DefaultRedirectHandler.java

```diff
@@ -0,0 +1,71 @@
+package io.vertx.core.http;
+
+import static io.vertx.core.http.HttpHeaders.CONTENT_LENGTH;
+import io.vertx.core.Future;
+import io.vertx.core.http.impl.HttpUtils;
+import java.net.URI;
+import java.util.function.Function;
+
+
+class DefaultRedirectHandler implements Function<HttpClientResponse, Future<RequestOptions>> {
```

Comment:
I prefer to keep this in impl even though it is public, it is part of implementation and JPMS makes it encapsulated